### PR TITLE
Clarify Ingress awaiter success condition and improve error message

### LINKED ISSUE #290: Update Ingress awaiter status message to be more clear on the success condition
As seen in #277, the reported status/errors for Ingress creation are not clear about the success conditions. Update the status message to clarify, and add a warning to check ingress-controller configuration (traefik does not set the expected status by default).

### PR DESCRIPTION
**Pull request by lblackstone**
Fixes #290  

Here's what it looks like after the fix:
screen shot 2018-11-21 at 11 18 00 am
screen shot 2018-11-21 at 11 19 05 am

### CODE REVIEW

File: pkg/await/extensions_ingress.go [hunk 1]
```diff
@@ -327,7 +327,9 @@ func (iia *ingressInitAwaiter) errorMessages() []string {
 
 	if !iia.ingressReady {
 		messages = append(messages,
-			"Ingress was not allocated an IP address; does your cloud provider support this?")
+			"Ingress .status.loadBalancer field was not updated with a hostname/IP address; "+
+				"If you're using the Traefik ingress-controller, make sure it is configured with the "+
```
Comment (lukehoban) [inline pkg/await/extensions_ingress.go]: While helpful, I'm not sure we should be guessing about specific causes like this unless we think those are 90% cases.  Can we generalize this at all to make it suggestive of the same root cause and fix, but while applying a larger class of causes?  Or should we start maintaining a Kubernetes provider troubleshooting docs page where we can include this kind of helpful text in longer form and link to it from the error messages here?

Comment (lukehoban) [reply pkg/await/extensions_ingress.go]: Curious @hausdorff thoughts on this as well.

Comment (lukehoban) [reply pkg/await/extensions_ingress.go]: For example - could we add something like "Ensure that your ingress controller is configured to update the `status` section of ingress objects"?  (Inspired by the text at https://docs.traefik.io/configuration/backends/kubernetes/)

Comment (lblackstone) [reply pkg/await/extensions_ingress.go]: Yeah, that's reasonable. One of the previous platforms I worked on would put a link in the error message to a knowledge base page we maintained. That way you could still get detailed troubleshooting info like that, but it was easier to maintain.

Comment (lukehoban) [reply pkg/await/extensions_ingress.go]: Yeah - we have https://pulumi.io/reference/troubleshooting.html which is used for this purpose by the CLI for core CLI errors.  We could extend that to have sections for other providers to add error message help as well, or could create dedicated troubleshooting pages under the `/kubernetes` section.

Comment (hausdorff) [reply pkg/await/extensions_ingress.go]: It sounds like we all agree about what to do here -- but to make it explicit, my preference is also to make error messages like this one follow a pattern:

* Factual explanation of what is wrong (_i.e._, ingress controller not configured to emit status)
* Link to a docs/playbooks/whatever

There are many reasons why I think this is a useful pattern -- lets us evolve recommended mitigations gradually and completely independently as we mature; not tractable to explain all mitigations in the error message; docs site experience is going to be way better for most complex issues; _etc_.

**Comment by lblackstone** [comment]
I've opened https://github.com/pulumi/docs/pull/672 and am in the process of registering a URL shortening service for `pulumi.info`. Once those are tasks are done, I'll update the message with a link pointing to those docs as suggested.

**Comment by lblackstone** [comment]
PR to merge doc changes to the prod website: https://github.com/pulumi/docs/pull/685

**Comment by lblackstone** [comment]
Here's the updated error:
image

**Comment by lblackstone** [comment]
Ok, this is ready to go. Here's the working shortlink: pulumi.io/help/k8s-ingress-lbstatus

**Comment by lblackstone** [comment]
@hausdorff Ah, I didn't realize we were already doing this elsewhere. I'll update to look more like https://github.com/pulumi/pulumi-service/blob/507e45bf80b8b6e0b744cc34e446ea04eb703430/cmd/service/services/cloud/managed_backend_updates.go#L114

FWIW, ctrl-click does work with the angle brackets in iterm2.

**Comment by lblackstone** [comment]
Opened https://github.com/pulumi/docs/pull/689 and will push an updated link here in a few mins.

**Comment by lblackstone** [comment]
image

**Pull request merged**